content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md

```diff
@@ -0,0 +1,113 @@
+
+<figure>
+	<img src="/files/blog/calisto-mathias/Final-Report/old-find-panel.png" alt="Old Find panel">
+	<center><figcaption>Old Find Panel</figcaption></center>
+</figure>
+
+<figure>
+	<img src="/files/blog/calisto-mathias/Final-Report/new-find-panel.png" alt="New Find Panel" width="598px">
+	<center><figcaption>New Find Panel</figcaption></center>
+</figure>
+
+### Adding Folder Filtering Abilities to Searches Executed Through the Find Panel
+
+The second task involved adding folder-based filtering to searches via the Find Panel. This enhancement was implemented at the application layer rather than the file system level. It maintains high search speeds while limiting results to selected directories. Users can now select multiple directories and merge the search results within the results panel. This functionality is accessible through the third drop-down menu from the left.
+
+Users can select directories using the last option in this drop-down menu. This action opens a file panel where they can choose one or more directories (or symbolic links to directories). The selected directories then appear in a clearly visible section within the same drop-down menu. Users also have the option to unselect any directories they no longer want to include.
+
+<figure>
+	<img src="/files/blog/calisto-mathias/Final-Report/find-panel-with-directory-selectors.png" alt="New Find Panel Directory Selector" width="598px">
+	<center><figcaption>Selecting Directories in the New Find Panel</figcaption></center>
+</figure>
+
+<figure>
+	<img src="/files/blog/calisto-mathias/Final-Report/Selecting-Directories.png" alt="New Find Panel Directory Selector" width="598px">
+	<center><figcaption>Selecting Directories in the New Find Panel (contd.)</figcaption></center>
+</figure>
+
+<figure>
+	<img src="/files/blog/calisto-mathias/Final-Report/directories-entries-in-menu.png" alt="New Find Panel Directory Selector" width="598px">
+	<center><figcaption>Selecting Directories in the New Find Panel (contd.)</figcaption></center>
+</figure>
+
+### Adding the Design for an Incremental Search Find Panel
+
+The most crucial and challenging aspect of my project was implementing the design for the incremental search find panel, which associates attribute columns with their corresponding search columns. While I managed to implement the core functionality, there are a few limitations that need to be addressed in the future.
+
+Currently, the incremental find panel integrated into the Find Panel results window is under review on GitHub (and soon on Gerrit). It will take some time before it is merged into the main codebase. Despite the challenges, I managed to get the core functionality working, allowing users to pause and restart their queries, as well as adjust search terms, all within the results window without switching back and forth repeatedly.
```

Comment:
Currently under review on Gerrit, right? You can link to the PR

Reply:
Yup. I had linked it at the bottom. I also added it as a link in that section now. I hope that's fine.